sdks/java/managed/src/main/java/org/apache/beam/sdk/managed/Managed.java

```diff
@@ -0,0 +1,253 @@
+              .build();
+
+      SchemaTransform underlyingTransform =
+          new ManagedSchemaTransformProvider(getSupportedIdentifiers()).from(managedConfig);
+
+      return input.apply(underlyingTransform);
+    }
+  }
+
+  /**
+   * Instantiates a {@link Managed.Write} transform for the specified sink. The supported managed
+   * sinks are:
+   *
+   * <ul>
+   *   <li>{@link Managed#ICEBERG} : Write to Apache Iceberg
+   * </ul>
+   */
+  public static Write write(String sink) {
+    return new AutoValue_Managed_Write.Builder()
+        .setSink(
+            Preconditions.checkNotNull(
+                Write.TRANSFORMS.get(sink.toLowerCase()),
+                "An unsupported sink was specified: '%s'. Please specify one of the following sinks: %s",
+                sink,
+                Write.TRANSFORMS.keySet()))
+        .setSupportedIdentifiers(new ArrayList<>(Write.TRANSFORMS.values()))
+        .build();
+  }
+
+  @AutoValue
+  public abstract static class Write extends SchemaTransform {
+    public static final Map<String, String> TRANSFORMS =
+        ImmutableMap.<String, String>builder()
+            .put(ICEBERG, "beam:schematransform:org.apache.beam:iceberg_write:v1")
+            .build();
+
+    abstract String getSink();
+
+    abstract @Nullable String getConfig();
+
+    abstract @Nullable String getConfigUrl();
+
+    abstract List<String> getSupportedIdentifiers();
+
+    abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setSink(String sinkIdentifier);
+
+      abstract Builder setConfig(@Nullable String config);
+
+      abstract Builder setConfigUrl(@Nullable String configUrl);
+
+      abstract Builder setSupportedIdentifiers(List<String> supportedIdentifiers);
+
+      abstract Write build();
+    }
+
+    /**
+     * Use the input Map of configuration arguments to build and instantiate the underlying sink.
+     * The map can ignore nullable parameters, but needs to include all required parameters. Check
+     * the underlying sink's configuration schema to see which parameters are available.
+     */
+    public Write withConfig(Map<String, Object> config) {
+      return toBuilder().setConfig(YamlUtils.yamlStringFromMap(config)).build();
+    }
+
+    /**
+     * Like {@link #withConfig(Map)}, but instead extracts the configuration arguments from a
+     * specified YAML file location.
+     */
+    public Write withConfigUrl(String configUrl) {
+      return toBuilder().setConfigUrl(configUrl).build();
+    }
+
+    @VisibleForTesting
+    Write withSupportedIdentifiers(List<String> supportedIdentifiers) {
+      return toBuilder().setSupportedIdentifiers(supportedIdentifiers).build();
+    }
+
+    @Override
+    public PCollectionRowTuple expand(PCollectionRowTuple input) {
+      ManagedSchemaTransformProvider.ManagedConfig managedConfig =
+          ManagedSchemaTransformProvider.ManagedConfig.builder()
```

Comment:
In fact, it looks like you already have the generic on here. I don't think the ephemeral Read/Write classes hold their weight just to have a `getSource()` and `getSink()` method instead of, say, `getIdentifier()`.

Reply:
Cham and I talked about this briefly in a [comment above](https://github.com/apache/beam/pull/30808#discussion_r1546828621). The concern was there may be reason for the Read and Write APIs to differ in the future.

FWIW I'm +1 on unifying to have the one API

Reply:
I think Robert's suggestion was to keep `Managed.read()` and `Managed.write()` as constructors but merge `Read` and `Write` classes (so the API shouldn't change). I'm +1 to that.

Reply:
And will return a single class `ManagedTransform` instead of `Read` and `Write`.

Reply:
Yep, thanks for clarifying that. Just committed those changes. @chamikaramj @robertwb let me know if this is what you had in mind